framework/contracts/account/tests/xion.rs

```diff
@@ -255,3 +255,269 @@ fn execute_from_res(deps: DepsMut, env: &Env, res: Response) -> AccountResult<Re
+    }
+
+    fn xion_wallet() -> anyhow::Result<xionrs::crypto::secp256k1::SigningKey> {
+        let daemon = Daemon::builder(XION_TESTNET_1)
+            .mnemonic(LOCAL_MNEMONIC)
+            .build()?;
+
+        let signing_key = xionrs::crypto::secp256k1::SigningKey::from_slice(
+            &daemon.sender().private_key.raw_key(),
+        )
+        .unwrap();
+
+        Ok(signing_key)
+    }
+
+    fn create_xion_account<Chain: CwEnv>(
+        abstr: &Abstract<Chain>,
+    ) -> anyhow::Result<AccountI<Chain>> {
+        let chain = abstr.registry.environment().clone();
+
+        // Generate salt from account id(or)
+        let salt = generate_instantiate_salt(&AccountId::local(
+            chain
+                .wasm_querier()
+                .item_query(&abstr.registry.address()?, LOCAL_ACCOUNT_SEQUENCE)?,
+        ));
+        let code_id = abstr.account_code_id().unwrap();
+
+        let account_addr = account_addr(&abstr)?;
+
+        let wallet = xion_wallet()?;
+        let signature = wallet.sign(account_addr.as_bytes()).unwrap();
+
+        chain
+            .instantiate2(
+                code_id,
+                &InstantiateMsg {
+                    code_id,
+                    account_id: None,
+                    owner: GovernanceDetails::AbstractAccount {
+                        address: account_addr.clone(),
+                    },
+                    namespace: None,
+                    install_modules: vec![],
+                    name: None,
+                    description: None,
+                    link: None,
+                    authenticator: Some(AddAuthenticator::Secp256K1 {
+                        id: 1,
+                        pubkey: Binary::new(wallet.public_key().to_bytes()),
+                        signature: Binary::new(signature.to_vec()),
+                    }),
+                },
+                Some("Abstract Account"),
+                Some(&account_addr),
+                &[],
+                salt,
+            )
+            .map_err(Into::into)?;
+
+        let account_id = chain
+            .wasm_querier()
+            .item_query(&account_addr, state::ACCOUNT_ID)?;
+        let contract_id = format!("{ACCOUNT}-{account_id}");
+
+        let account = AccountI::new(contract_id, chain);
+        account.set_address(&account_addr);
+        Ok(account)
+    }
+
+    #[test]
+    #[serial_test::serial]
+    fn xion_account_creation() -> anyhow::Result<()> {
+        let mock = MockBech32::new("xion");
+
+        let abstr = Abstract::deploy_on(mock, ())?;
+
+        // We create an XION abstract account
+        create_xion_account(&abstr)?;
+
+        Ok(())
+    }
+
+    fn before_hook(
+        abstr: &Abstract<MockBech32>,
+        account: &AccountI<MockBech32>,
+    ) -> anyhow::Result<()> {
+        let mock = abstr.registry.environment().clone();
+        let sign_doc_bytes = Binary::from(HexBinary::from_hex("a527761bf3e9279be8cf")?);
```

Comment:
Can we add a comment with the source of this hex

Reply:
This is random hex (adding a comment)